Cannot compile using sudo docker-compose up

### Overview / TLDR
When running sudo docker-compose up, the documentation leads me to believe this is all I need to do to compile.

However I'm getting two distinct errors out the gate. I've included details below with headings above each section. Straight up, if I've missed something obvious, please let me know.

### Commands exactly as entered
git clone https://github.com/dectalk/dectalk.git
cd dectalk
sudo docker-compose up

### Command Output
WARN[0000] /home/***/Documents/Code/dectalk/docker-compose.yml: `version` is obsolete 
[+] Running 1/0
 ✔ Container dectalk-dectalk-1  Created                                                                                                                                                                                                                                                                                  0.0s 
Attaching to dectalk-1
dectalk-1  | fatal: detected dubious ownership in repository at '/dectalk'
dectalk-1  | To add an exception for this directory, call:
dectalk-1  | 
dectalk-1  | 	git config --global --add safe.directory /dectalk
dectalk-1  | fatal: detected dubious ownership in repository at '/dectalk'
dectalk-1  | To add an exception for this directory, call:
dectalk-1  | 
dectalk-1  | 	git config --global --add safe.directory /dectalk
dectalk-1  | fatal: detected dubious ownership in repository at '/dectalk'
dectalk-1  | To add an exception for this directory, call:
dectalk-1  | 
dectalk-1  | 	git config --global --add safe.directory /dectalk
dectalk-1  | fatal: detected dubious ownership in repository at '/dectalk'
dectalk-1  | To add an exception for this directory, call:
dectalk-1  | 
dectalk-1  | 	git config --global --add safe.directory /dectalk
dectalk-1  | configure: error: cannot find sources (DECtalk) in . or ..
dectalk-1 exited with code 1

### Attempted workarounds
I've run the suggested git config command outside the container, which seems to repress the ownership warning.

Re the second error about not finding the sources in . or .., I've manually loaded the container with:
docker run -it -v ./:/dectalk dectalk-dectalk:latest bash

With the shell loaded, I confirmed the source directory was mounted in /dectalk, and executed the commands manually as written in the dockerfile. autoreconf ran with no visible output, and ./configure returned the same error of no source in . or .., see above.

I then tried the same approach but used chown root:root /dectalk -R beforehand, just in case. I've also tried the same with setting permissions to 777 on the whole repo on the host, as an extreeme measure.

All these have returned the same output as included above.

### Closeing thoughts

I'm happy to put in some time to help solve this, my linux experience is definitely beyond my limited programming knowledge but I'll do what I can.

```
docker-compose up
Building dectalk
[+] Building 0.6s (8/8) FINISHED                                                                                 docker:default
 => [internal] load build definition from Dockerfile                                                                       0.0s
 => => transferring dockerfile: 277B                                                                                       0.0s
 => [internal] load metadata for docker.io/library/debian:bullseye                                                         0.6s
 => [internal] load .dockerignore                                                                                          0.0s
 => => transferring context: 2B                                                                                            0.0s
 => [1/4] FROM docker.io/library/debian:bullseye@sha256:37d7709beef67abbccfcac94a2150d25fdac3764251a60f2c20e9fd069d228d8   0.0s
 => CACHED [2/4] RUN apt-get update                                                                                        0.0s
 => CACHED [3/4] RUN apt-get --no-install-recommends -y install build-essential libasound2-dev libpulse-dev libgtk2.0-dev  0.0s
 => CACHED [4/4] WORKDIR /dectalk/src                                                                                      0.0s
 => exporting to image                                                                                                     0.0s
 => => exporting layers                                                                                                    0.0s
 => => writing image sha256:87bf5046561454f8af54c95e3d9fd54efba04a87cc381f6966ebadefdda0bf68                               0.0s
 => => naming to docker.io/library/dectalk_dectalk                                                                         0.0s

 1 warning found (use docker --debug to expand):
 - JSONArgsRecommended: JSON arguments recommended for CMD to prevent unintended behavior related to OS signals (line 8)
WARNING: Image for service dectalk was built because it did not already exist. To rebuild this image you must use `docker-compose build` or `docker-compose up --build`.
Creating dectalk_dectalk_1 ... done
Attaching to dectalk_dectalk_1
dectalk_1  | configure: error: cannot find sources (DECtalk) in . or ..
dectalk_dectalk_1 exited with code 1
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,8 +1,17 @@
-FROM debian:bullseye
+FROM debian:bullseye AS build
 
 RUN apt-get update
 RUN apt-get --no-install-recommends -y install build-essential libasound2-dev libpulse-dev libgtk2.0-dev unzip autoconf automake git
 
+COPY . /dectalk
+
 WORKDIR /dectalk/src
 
-CMD autoreconf -i && ./configure && make
+RUN ./autogen.sh
+RUN ./configure
+RUN make -j
+
+FROM scratch AS output-only
+
+COPY --from=build /dectalk/dist ./dist
+

--- a/README.md
+++ b/README.md
@@ -58,7 +58,9 @@ The built files will be found in the `/dist` folder.
 <summary>Compiling on Linux with Docker</summary>
 
 To build DECtalk without setting up a local build environment, run `sudo docker-compose up`
-(and make sure you have Docker and docker-compose installed!)
+(and make sure you have Docker and docker-compose installed!). The `dist` directory should appear in the project root with the build output.
+
+If you want to build with only Docker, you can run `sudo docker build --output=. .` to build and output the `dist` directory to the project root. If you run into permission errors, try making a new directory with full write access and output to it, e.g. `mkdir output`, `chmod 777 ./output`, `sudo docker build --output=./output .`.
 
 </details>
 

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -1,7 +1,8 @@
-version: '3.6'
 services:
   dectalk:
-    build: .
+    build:
+      context: .
+      target: build
     volumes:
-      - ./:/dectalk
-      
+      - ./dist:/dist
+    command: ["cp", "-r", "/dectalk/dist/.", "/dist"]

